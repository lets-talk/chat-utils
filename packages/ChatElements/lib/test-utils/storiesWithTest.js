import { withTests } from '@storybook/addon-jest';
import jestTestResults from '../../tests-results/jest-test-results.json';

export default withTests(jestTestResults, {
  filesExt: '__tests__/**.js',
});
