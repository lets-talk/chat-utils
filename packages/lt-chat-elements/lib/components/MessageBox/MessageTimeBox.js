import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import moment from 'moment';

// Icons to be used on this component
import IoDoneAll from 'react-icons/lib/io/android-done-all';
import MdIosTime from 'react-icons/lib/md/access-time';
import MdCheck from 'react-icons/lib/md/check';

// Constants
import constants from '../../utils/constants';
import { textColor } from '../../utils/style';

const { messagesTypes } = constants;

const StyledMessageTimeBoxContainer = styled.div`
  max-width: 80px;
  text-align: right;
  color: ${(props) => textColor(props.theme, 'light', 'disabled')};
  font-size: ${(props) => props.theme.typography.classes.caption.fontSize};
  line-height: ${(props) => props.theme.typography.classes.caption.lineHeight};
  font-weight: ${(props) => props.theme.typography.weights.fontWeightMedium};
  float: right
  position: absolute;
  bottom: 0;
  right: 0;
`;
StyledMessageTimeBoxContainer.displayName = 'StyledMessageTimeBoxContainer';

const StyledTimeAgoText = styled.span`
  font-size: ${(props) => props.theme.typography.classes.small.fontSize};
`;
StyledTimeAgoText.displayName = 'StyledTimeAgoText';

const StyledStatusSpan = styled.span`
  margin-left: 3px;
`;
StyledStatusSpan.displayName = 'StyledStatusSpan';

const MessageTimeBox = (props) => (
  <StyledMessageTimeBoxContainer>
    {
      props.date && (props.type !== messagesTypes.TYPING) &&
      <StyledTimeAgoText>
        {props.dateString || moment(props.date).fromNow()}
      </StyledTimeAgoText>
    }
    {
      props.status &&
        <StyledStatusSpan>
          {
            props.status === 'waiting' &&
            <MdIosTime />
          }

          {
            props.status === 'sent' &&
            <MdCheck />
          }

          {
            props.status === 'received' &&
            <IoDoneAll />
          }

          {
            props.status === 'read' &&
            <IoDoneAll color="#4FC3F7" />
          }
        </StyledStatusSpan>
    }
  </StyledMessageTimeBoxContainer>
);

MessageTimeBox.propTypes = {
  /**
   * Type of Message: Type of the message. Values are defined as constants.
   * The current supported types are: SYSTEM, TEXT, TYPING, TIME, ACTIONABLE, FILE, PHOTO.
   */
  type: PropTypes.string.isRequired,
  /**
   * status: Message status. Values are defined as constants.
   * Current supported values are: WAITING, SENT, RECEIVED, READ
   */
  status: PropTypes.string,
  /**
   * date: Message creation date.
   */
  date: PropTypes.instanceOf(Date),
  /**
   * dateString: Formated date string to show. This is the default to use.
   * If this is not provided by default to show the time moment(props.date).fromNow() is used.
   */
  dateString: PropTypes.string,
};

MessageTimeBox.defaultProps = {
  status: '',
  date: null,
  dateString: '',
};

export default MessageTimeBox;
