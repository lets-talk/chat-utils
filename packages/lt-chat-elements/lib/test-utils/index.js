// test-utils/index.js

import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';
import { ThemeProvider } from 'styled-components';

const defaultTheme = {
  palette: {
    common: {
      black: 'rgb(0, 0, 0)',
      white: 'rgb(255, 255, 255)',
      online: 'rgb(67, 160, 71)',
      offline: 'rgb(229, 57, 53)',
      sleeping: 'rgb(253, 216, 53)',
    },
    colors: {
      primary: {
        base: 'rgb(4, 38, 73)',
        light: 'rgb(86, 165, 246)',
        dark: 'rgb(0, 0, 0)',
      },
      accent: {
        base: 'rgb(255, 87, 50)',
        light: 'rgb(255, 171, 152)',
        dark: 'rgb(203, 37, 0)',
      },
      foreground: {
        base: 'rgb(94, 124, 139)',
        light: 'rgb(151, 173, 184)',
        dark: 'rgb(53, 70, 78)',
      },
      background: {
        base: 'rgb(255, 255, 255)',
        light: 'rgb(255, 255, 255)',
        dark: 'rgb(204, 204, 204)',
      },
    },
    contrastColors: {
      dark: {
        primary: 'rgb(255, 255, 255)',
        secondary: 'rgba(255, 255, 255, 0.7)',
        disabled: 'rgba(255, 255, 255, 0.5)',
        hint: 'rgba(255, 255, 255, 0.12)',
      },
      light: {
        primary: 'rgba(0, 0, 0, 0.87)',
        secondary: 'rgba(0, 0, 0, 0.54)',
        disabled: 'rgba(0, 0, 0, 0.38)',
        hint: 'rgba(0, 0, 0, 0.12)',
      },
    },
    error: {
      light: '#e57373',
      main: '#f44336',
      dark: '#d32f2f',
      contrastText: '#fff',
    },
    divider: 'rgba(255, 255, 255, 0.1)',
    action: {
      active: 'rgba(255, 255, 255, 0.9)',
      hover: 'rgba(255, 255, 255, 0.08)',
      selected: 'rgba(255, 255, 255, 0.14)',
      disabled: 'rgba(255, 255, 255, 0.26)',
      disabledBackground: 'rgba(255, 255, 255, 0.12)',
    },
  },
  shadows: {
    '2p': '0 2px 2px 0 rgba(0, 0, 0, 0.14), 0 3px 1px -2px rgba(0, 0, 0, 0.2), 0 1px 5px 0 rgba(0, 0, 0, 0.12)',
    '3p': '0 3px 4px 0 rgba(0, 0, 0, 0.14), 0 3px 3px -2px rgba(0, 0, 0, 0.2), 0 1px 8px 0 rgba(0, 0, 0, 0.12)',
    '4p': '0 4px 5px 0 rgba(0, 0, 0, 0.14), 0 1px 10px 0 rgba(0, 0, 0, 0.12), 0 2px 4px -1px rgba(0, 0, 0, 0.2)',
    '6p': '0 6px 10px 0 rgba(0, 0, 0, 0.14), 0 1px 18px 0 rgba(0, 0, 0, 0.12), 0 3px 5px -1px rgba(0, 0, 0, 0.2)',
    '8p': '0 8px 10px 1px rgba(0, 0, 0, 0.14), 0 3px 14px 2px rgba(0, 0, 0, 0.12), 0 5px 5px -3px rgba(0, 0, 0, 0.2)',
    '16p': '0 16px 24px 2px rgba(0, 0, 0, 0.14), 0 6px 30px 5px rgba(0, 0, 0, 0.12), 0 8px 10px -5px rgba(0, 0, 0, 0.2)',
  },
  typography: {
    fontFamily: "'Nimbus sans-serif', 'Sans', 'Helvetica', 'Arial'",
    fontSize: 'calc(1 * 1px)',
    weights: {
      fontWeightUltraThin: 100,
      fontWeightThin: 300,
      fontWeightNormal: 300,
      fontWeightMedium: 500,
      fontWeightSemibold: 600,
      fontWeightBold: 700,
    },
    classes: {
      body: {
        fontSize: 'calc(16 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      caption: {
        fontSize: 'calc(12 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      footnote: {
        fontSize: 'calc(14 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      small: {
        fontSize: 'calc(10 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      headline: {
        fontSize: 'calc(24 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      subhead: {
        fontSize: 'calc(18 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      title: {
        fontSize: 'calc(20 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      jumbo: {
        fontSize: 'calc(44 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      mega: {
        fontSize: 'calc(56 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
    },
  },
  components: {
    conversationBox: {
      externalBackgroundColor: 'rgba(94, 124, 139, 0.1)',
      internalBackgroundColor: 'rgba(255, 213, 0, 0.2)',
      importantBackgroundColor: 'rgb(255, 246, 244)',
    },
    loader: {
      backgroundColor: 'rgb(252, 88, 48)',
    },
  },
};

const shallowWithTheme = (component) => {
  const context = shallow(<ThemeProvider theme={defaultTheme} />)
    .instance()
    .getChildContext();
  return shallow(component, {
    context,
    childContextTypes: ThemeProvider.childContextTypes,
  });
};

const mountWithTheme = (component) => {
  const context = shallow(<ThemeProvider theme={defaultTheme} />)
    .instance()
    .getChildContext();

  return mount(component, {
    context,
    childContextTypes: ThemeProvider.childContextTypes,
  });
};

const renderWithTheme = (component) => renderer.create(<ThemeProvider theme={defaultTheme}>{component}</ThemeProvider>);

const staticWithTheme = (component) => render(<ThemeProvider theme={defaultTheme}>{component}</ThemeProvider>);

export {
  defaultTheme,
  shallowWithTheme,
  mountWithTheme,
  staticWithTheme,
  renderWithTheme,
};
