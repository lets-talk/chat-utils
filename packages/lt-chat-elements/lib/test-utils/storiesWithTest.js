import { withTests } from '@storybook/addon-jest';
import jestTestResults from '../../tests-results/jest-test-results.json';

export default withTests({ results: jestTestResults }, {
  filesExt: '__tests__/**.js',
});
