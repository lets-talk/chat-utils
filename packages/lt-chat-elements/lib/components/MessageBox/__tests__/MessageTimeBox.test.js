import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import 'jest-styled-components';
import MessageTimeBox from '../MessageTimeBox';
// Constants
import { renderWithTheme } from '../../../test-utils';

describe('MessageTimeBox component', () => {
  it('should render a Text Message and contain the Text and status', () => {
    const props = {
      type: 'text',
      status: 'waiting',
      date: new Date('2018-06-10 22:22:02'),
    };

    const component = shallow(<MessageTimeBox {...props} />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
    expect(component.find('StyledMessageTimeBoxContainer').length).toBe(1);
  });


  it('should render a Text Message and contain the Text and status', () => {
    const props = {
      type: 'text',
      status: 'received',
      date: new Date('2018-06-10 22:22:02'),
    };

    const component = shallow(<MessageTimeBox {...props} />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
    expect(component.find('StyledMessageTimeBoxContainer').length).toBe(1);
  });

  test('Styled: Should Render expected Styles when position is left', () => {
    const props = {
      type: 'text',
      position: 'left',
      status: 'received',
      date: new Date('2018-06-10 22:22:02'),
    };
    const tree = renderWithTheme(<MessageTimeBox {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
