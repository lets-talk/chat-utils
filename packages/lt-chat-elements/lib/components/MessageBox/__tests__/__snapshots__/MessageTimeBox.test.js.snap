// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageTimeBox component Styled: Should Render expected Styles when position is left 1`] = `
.c0 {
  max-width: 80px;
  text-align: right;
  color: rgba(0,0,0,0.38);
  font-size: calc(12 * 1px);
  line-height: calc(16 * 1px);
  font-weight: 500;
  float: right;
  position: absolute;
  bottom: 0;
  right: 0;
}

.c1 {
  font-size: calc(10 * 1px);
}

.c2 {
  margin-left: 3px;
}

<div
  className="c0"
>
  <span
    className="c1"
  >
    4 days ago
  </span>
  <span
    className="c2"
  >
    <svg
      fill="currentColor"
      height="1em"
      preserveAspectRatio="xMidYMid meet"
      style={
        Object {
          "color": undefined,
          "verticalAlign": "middle",
        }
      }
      viewBox="0 0 40 40"
      width="1em"
    >
      <g>
        <path
          d="m30.3 10.9l-10.9 10.9-2.4-2.4 10.9-10.9z m7.3-2.4l2.4 2.4-20.6 20.6-9.6-9.6 2.4-2.4 7.2 7.1z m-37.6 13.4l2.5-2.4 9.5 9.6-2.4 2.4z"
        />
      </g>
    </svg>
  </span>
</div>
`;

exports[`MessageTimeBox component should render a Text Message and contain the Text and status 1`] = `
<StyledMessageTimeBoxContainer>
  <StyledTimeAgoText>
    4 days ago
  </StyledTimeAgoText>
  <StyledStatusSpan>
    <MdAccessTime />
  </StyledStatusSpan>
</StyledMessageTimeBoxContainer>
`;

exports[`MessageTimeBox component should render a Text Message and contain the Text and status 2`] = `
<StyledMessageTimeBoxContainer>
  <StyledTimeAgoText>
    4 days ago
  </StyledTimeAgoText>
  <StyledStatusSpan>
    <IoAndroidDoneAll />
  </StyledStatusSpan>
</StyledMessageTimeBoxContainer>
`;
